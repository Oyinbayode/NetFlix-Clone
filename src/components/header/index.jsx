import React from "react";
import { Link as ReactRouterLink } from "react-router-dom";

import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  FeatureCallOut,
  Text,
  TLink,
  Group,
  Picture,
  Profile,
  Dropdown,
} from "./styles/header";

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <TLink {...restProps}>{children}</TLink>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.ButtonLink = function HeaderButtonLink({ to, children, restProps }) {
  return (
    <ButtonLink to={to} {...restProps}>
      {children}
    </ButtonLink>
  );
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};
