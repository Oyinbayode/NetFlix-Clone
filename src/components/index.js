export { default as Jumbotron } from "./jumbotron";
export { default as Footer } from "./footer";
export { default as Accordion } from "./accordion";
export { default as OptForm } from "./opt-form";
export { default as Header } from "./header";
export { default as Feature } from "./feature";
export { default as SigninForm } from "./sign-in-form";
export { default as Profiles } from "./profiles";
export { default as Loading } from "./loading";
export { default as Card } from "./card";
export { default as Player } from "./player";
