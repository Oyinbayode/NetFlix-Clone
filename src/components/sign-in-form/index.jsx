import React from "react";
import {
  Container,
  Error,
  Base,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from "./styles/sign-in-form";

export default function SigninForm({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

SigninForm.Error = function SigninFormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

SigninForm.Base = function SigninFormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

SigninForm.Title = function SigninFormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

SigninForm.Text = function SigninFormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

SigninForm.TextSmall = function SigninFormTextSmall({
  children,
  ...restProps
}) {
  return <TextSmall {...restProps}>{children}</TextSmall>;
};

SigninForm.Link = function SigninFormLink({ to, children, ...restProps }) {
  return (
    <Link to={to} {...restProps}>
      {children}
    </Link>
  );
};

SigninForm.Input = function SigninFormInput({ ...restProps }) {
  return <Input {...restProps} />;
};

SigninForm.Submit = function SigninFormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};
