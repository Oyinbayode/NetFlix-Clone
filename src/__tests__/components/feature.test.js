import React from "react";
import { render, screen } from "@testing-library/react";
import { Feature } from "../../components";

describe("<Feature />", () => {
  it("renders the <Feature /> with populated data", () => {
    const { container } = render(
      <Feature>
        <Feature.Title>Unlimited movies, TV shows, and more.</Feature.Title>
        <Feature.SubTitle>Watch anywhere. Cancel anytime.</Feature.SubTitle>
      </Feature>
    );

    expect(
      screen.getByText("Unlimited movies, TV shows, and more.")
    ).toBeTruthy();
    expect(screen.getByText("Watch anywhere. Cancel anytime.")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Feature /> with title", () => {
    const { container } = render(
      <Feature>
        <Feature.Title>Unlimited movies, TV shows, and more.</Feature.Title>
      </Feature>
    );

    expect(
      screen.getByText("Unlimited movies, TV shows, and more.")
    ).toBeTruthy();
    expect(screen.queryByText("Watch anywhere. Cancel anytime.")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Feature /> with subtitle", () => {
    const { container } = render(
      <Feature>
        <Feature.SubTitle>Watch anywhere. Cancel anytime.</Feature.SubTitle>
      </Feature>
    );

    expect(
      screen.queryByText("Unlimited movies, TV shows, and more.")
    ).toBeFalsy();
    expect(screen.getByText("Watch anywhere. Cancel anytime.")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
