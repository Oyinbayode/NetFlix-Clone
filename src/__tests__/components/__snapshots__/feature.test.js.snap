// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders the <Feature /> with populated data 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 kXOooJ"
>
  <h1
    class="feature__Title-sc-6swrp6-1 dtcvs"
  >
    Unlimited movies, TV shows, and more.
  </h1>
  <h2
    class="feature__SubTitle-sc-6swrp6-2 hVoHew"
  >
    Watch anywhere. Cancel anytime.
  </h2>
</div>
`;

exports[`<Feature /> renders the <Feature /> with subtitle 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 kXOooJ"
>
  <h2
    class="feature__SubTitle-sc-6swrp6-2 hVoHew"
  >
    Watch anywhere. Cancel anytime.
  </h2>
</div>
`;

exports[`<Feature /> renders the <Feature /> with title 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 kXOooJ"
>
  <h1
    class="feature__Title-sc-6swrp6-1 dtcvs"
  >
    Unlimited movies, TV shows, and more.
  </h1>
</div>
`;
